tests/ui/frontmatter/frontmatter-after-tokens.rs

```diff
@@ -0,0 +1,8 @@
+#![feature(frontmatter)]
+
+---
+//~^ ERROR: expected item, found `-`
+// FIXME(frontmatter): make this diagnostic better
+---
```

Comment:
Remark: yeah, I'm not sure how to detect this properly without making some convoluted logic, even though I imagine this would be one of the more common mistakes to make. At least this does error, *maybe* we could provide some kind of contextual HELP like if we see `---` (3+ starting dashes)

```
error: expected item, found `-`
  --> $DIR/frontmatter-after-tokens.rs:3:1
   |
LL | ---
   | ^ expected item
   |
   = note: for a full list of items that can appear in modules, see <https://doc.rust-lang.org/reference/items.html>
   = help: if you meant to write a frontmatter, the frontmatter must come after an optional shebang but before any regular source code
```

(With some better wording, I find it tricky to explain)

Reply:
Words are hard 🙂 

If the file doesn't start with a shebang line, maybe we could just say something like

```
   = help: if you meant to write a frontmatter, the frontmatter must appear first in the file
```

Shebangs are not super commonly used so if we can leave that detail out, it would simplify the error message.